writeaheadlog.go

```diff
@@ -242,6 +242,16 @@ func (w *WAL) recoverWAL(data []byte) ([]Update, error) {
 			firstPage.nextPage = &nextDiskPage.page
 		}
 
+		// Check if the pages of the transaction form a loop
+		visited := make(map[uint64]struct{})
+		for page := firstPage; page != nil; page = page.nextPage {
+			if _, exists := visited[page.offset]; exists {
+				// Loop detected
+				continue
```

Comment:
this `continue` is scoped to the inner `for` loop, but I think you want the outer one. Add a label like `nextTxn:` above the outer `for` loop and change this to `continue nextTxn`

Reply:
Oops you are right